include/crocoddyl/multibody/contacts/contact-3d.hpp

```diff
@@ -15,9 +15,12 @@
 #include <pinocchio/algorithm/kinematics-derivatives.hpp>
 
 #include "crocoddyl/multibody/fwd.hpp"
-#include "crocoddyl/multibody/frames.hpp"
 #include "crocoddyl/core/utils/exception.hpp"
 #include "crocoddyl/multibody/contact-base.hpp"
+#include "crocoddyl/core/utils/deprecate.hpp"
+
+#define IGNORE_DEPRECATED_HEADER  // TODO: Remove once the deprecated FrameXX has been removed in a future release
+#include "crocoddyl/multibody/frames.hpp"
```

Comment:
>     1. undefining the IGNORE_DEPRECTATED_HEADER so that it is not activated by default

I added the `undef` rule in the frames.hpp file. So, we impose every-time the need to print the deprecation message, and it is deactivated only if the developer deliberately define `CROCODDYL_IGNORE_DEPRECATED_HEADER`.



>     2\. Moving the IGNORE_DEPRECATED_HEADER to CROCODDYL_IGNORE_DEPRECATED_HEADER. We should also start moving macros to a dedicated file, but that could be handled later.

Note that `CROCODDYL_IGNORE_DEPRECATED_HEADER` is `undef` by default as we want to always activate the deprecation message. So, I believe we don't need to add `#undef CROCODDYL_IGNORE_DEPRECATED_HEADER` in a file dedicated to macros, right?

Is this what you asking? If so, then could you explain me why if this needed?



>     3\. Renaming the xref_, pref_, cref etc to simply ref. I don't think the user should have to remember which particular symbol we decided to use for a particular contact or cost class.

`xref_`, `pref_`, and the other references are protected variables. The user can retrieve does reference by
```c++
  contact.get_reference();  //c++ code
```
or
```python
  contact.reference ## Python code
```

We are doing the same for residual functions. Would you still have concerns with the names of the protected variables?

Reply:
No, we can manage the macros at a later date, and the same goes for the `xref_` etc values, it is just easier to go through the c++ code when the python code is closely aligned. However, that should not block this PR

Reply:
Regarding the undef macro, when user imports any file from multibody folder, he/she is bound to include the frame.hpp and thus activate this macro. It would be better that this is not the standard behaviour, we want to be able to re-use this macro for deprecations that we don't want warnings about, and don't want the user to suppress deprecations that we do want to warn him/her about. We can closely define this macro with the deprecation scheme `CROCODDYL_PRAGMA_DEPRECATED_HEADER`, but don't have to do it in this PR, this could be handled later

Reply:
Does this answer your question?

Reply:
A better solution might be to 
1) in ./include/crocoddyl/core/utils/deprecate.hpp, put `CROCODDYL_PRAGMA_DEPRECATED_HEADER` within the macro 
```c++
#ifdef CROCODDYL_IGNORE_DEPRECATED_HEADER
#define CROCODDYL_PRAGMA_DEPRECATED_HEADER(old_header, new_header)
...
#else
#define CROCODDYL_PRAGMA_DEPRECATED_HEADER(old_header, new_header)
//DoNothing
#endif
```

And then in frames.hpp
do 
```c++
#define CROCODDYL_IGNORE_DEPRECATED_HEADER
CROCODDYL_PRAGMA_DEPRECATED_HEADER(old_header, new_header)
#undef CROCODDYL_IGNORE_DEPRECATED_HEADER
```
And remove `CROCODDYL_IGNORE_DEPRECATED_HEADER` from all the other files that are calling frame.
This way, when we want to warn users about the frame deprecation, we can simply remove the structure `CROCODDYL_IGNORE_DEPRECATED_HEADER` in _one_ single file, and be done with it

Reply:
Could you the changes in this PR?

Reply:
This conversation is now resolved. Thanks Rohan!